### DISCUSSION TITLE
Fandom Wikis and Web Scraping Homework

### DISCUSSION THREAD

**Discussion by ZoeLeBlanc** [CultureAsData-UIUC/is310-fall-2024#9] [Assignments]
Please post the link to your folder for this assignment [https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/creating-curating-humanities-data/05-web-scraping#fandom-wikis-and-web-scraping-homework](https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/creating-curating-humanities-data/05-web-scraping#fandom-wikis-and-web-scraping-homework). Don't hesitate to reach out if you have questions!

**Comment by ryanpearlman1** [comment]
https://github.com/ryanpearlman1/is310-coding-assignments/tree/main/web_scraping_assignments

**Comment by yosefstifanos** [comment]
https://github.com/yosefstifanos/is310-coding-assignments/tree/main/web_scraping_assignments

**Comment by tinayang922** [comment]
https://github.com/tinayang922/is310-coding-assignments/tree/main/web_scraping_assignments

**Comment by kohtas2** [comment]
https://github.com/kohtas2/is310-coding-assignments/tree/main/web_scraping_assignments

**Comment by Cindy1226** [comment]
https://github.com/Cindy1226/is310-coding-assignments/tree/main/web_scraping_assignments

**Comment by shoirathi** [comment]
https://github.com/shoirathi/is310-coding-assignments/tree/main/web_scraping

**Comment by GloriaLiu930** [comment]
https://github.com/GloriaLiu930/is310-coding-assignments/tree/main/web_scraping_assignments

**Comment by elee33** [comment]
https://github.com/elee33/web_scraping_assignments

**Comment by divyaj5** [comment]
https://github.com/divyaj5/is310-coding-assignments/tree/master/web_scraping

**Comment by ahmadsawwan** [comment]
https://github.com/ahmadsawwan/ahmadsawwan-is310-coding-assignments/tree/main/web_scraping_assignments%20copy

**Reply by ZoeLeBlanc** [reply]
Hi @ahmadsawwan ! This is a strong start but noticed a few issues when I tried running your code. First, based on the error below seems like you aren't using `requests` but downloading an HTML file? 

<img width="953" alt="Screenshot 2024-11-11 at 10 26 56 PM" src="https://github.com/user-attachments/assets/da3b5f42-91a5-4630-91ca-48270820a374">

Is that because you decided to scrape a non-fandom wiki? I realize the TV guide website might be trickier to scrape so happy to help consult or you're welcome to choose an easier website. But hopefully you can just update your code to use requests and then you should be good to go 👍

**Reply by ahmadsawwan** [reply]
I fixed it now! Sorry for it being late. I get a notification for every feedback that happens in the class, so the ones that pertain to me get flooded and hidden by ones that are not for me, so I only found out about this when you put the grades in. Thank you!

**Comment by OctaneCodes** [comment]
https://github.com/OctaneCodes/is310-coding-assignments/tree/main/web_scraping_assignments

**Reply by ZoeLeBlanc** [reply]
Hi @OctaneCodes ! Very impressive script, and especially like the rich table––very nice touch and well done 👍

**Comment by averyWebster** [comment]
https://github.com/averyWebster/is310-coding-assignments/tree/main/web_scraping_assignments

**Comment by haydnelm** [comment]
https://github.com/haydnelm/is310-coding-assignments/tree/main/web-scraping

**Reply by ZoeLeBlanc** [reply]
Hi @haydnelm ! This is very impressive and while I have no knowledge of Runescape or its bosses, I'm really impressed with the scraping you're doing here. Only one small tweak request: you are not saving the data to file. Once you update the code to do that then you'll get full marks 👍

**Reply by haydnelm** [reply]
updated last week 👍

**Comment by jasonmorris8** [comment]
https://github.com/jasonmorris8/is310-coding-assignments/tree/main/web_scraping_assignments

**Comment by officalcharles** [comment]
https://github.com/officalcharles/is310-coding-assignments/tree/master/web_scraping_assignments

**Comment by SamuelSokolovsky** [comment]
https://github.com/SamuelSokolovsky/is310-coding-assignments/tree/main/web_scraping_assignments

**Comment by esr27** [comment]
https://github.com/esr27/is310-coding-assignments/tree/main/web_scraping_assigments

**Comment by lhayedimarucut** [comment]
https://github.com/lhayedimarucut/is310-coding-assignments/tree/main/web_scraping_assignments

**Reply by ZoeLeBlanc** [reply]
Hi @lhayedimarucut ! Just wanted to compliment you on your very nice script. Especially liked the rich table in the output, so very well done 👍

**Comment by Rebe-Rod** [comment]
https://github.com/Rebe-Rod/is310-coding-assingments/tree/main/web_scraping_assignments

**Reply by ZoeLeBlanc** [reply]
Hi @Rebe-Rod ! The code works great but just noticed you named your `README.md` as `REAMME.md`. Not a huge problem but wanted to flag it for you because that means your file doesn't render in GitHub properly. Good work overall though 👍

**Comment by LucasWashor** [comment]
https://github.com/LucasWashor/is310-coding-assignments/tree/main/web_scraping

**Comment by henRotunno** [comment]
https://github.com/henRotunno/is310-coding-assignments/tree/main/web_scraping_assignments

**Comment by adg77** [comment]
https://github.com/adg77/is310-coding-assignments/tree/main/web_scraping_assignments

**Reply by ZoeLeBlanc** [reply]
Hi @adg77 ! Apologies for the delayed response but your code looks great. There's just one small issue. You've hard coded your file path for writing the file which gives the following error:

<img width="958" alt="Screenshot 2024-11-11 at 9 07 03 PM" src="https://github.com/user-attachments/assets/4ffe90b9-022a-4b2a-b95c-9e281082e207">

To fix this, all you need to do is remove the parts of the file path that are specific to your computer. Happy to help if you have questions and once you fix that you'll receive full marks 👍

**Reply by adg77** [reply]
Sorry for the late fix, but should be fixed now. Thanks
https://github.com/adg77/is310-coding-assignments/blob/main/web_scraping_assignments/fandom.wiki.scraping.py

**Comment by chuzesta** [comment]
https://github.com/chuzesta/is310-coding-assignments/tree/main/web_scraping_assignments

**Reply by ZoeLeBlanc** [reply]
Hi @chuzesta ! This is very impressive and thorough web scraping so well done 👍 ! Just wanted to flag two small things for you. First it looks you may have accidentally created a file called `-m` in your root `is310-coding-assignments` folder. This isn't a problem at all, but I suspect you made it accidentally when trying to make a git commit. I would recommend deleting the file though again not a problem. The other is you are naming your files `readME.md`. Again this is purely stylistic advice, but usually we write it as `README.md`. The reason has to do with history of computing and Unix file systems so not really important any more but in case you cared, I just wanted to let you know the norm. I'm happy to respect whatever naming convention you choose and happy to clarify if anything above is confusing!

**Comment by frangara** [comment]
Attached is the link to the GitHub. I apologize for turning this in so late.

https://github.com/frangara/is310-coding-assignments/tree/main/web_scraping

**Comment by AG-Samaniego** [comment]
My submission: https://github.com/AG-Samaniego/is310-coding-assignments/tree/main/web-scraping
(Admittedly not as much metadata as I'd like but it's my min viable product right now; I'll probably play around with it more during break just for fun)

**Comment by yukthas2** [comment]
https://github.com/yukthas2/is310-coding-assignments/tree/main/web_scraping

**Comment by theajk2** [comment]
https://github.com/yukthas2/is310-coding-assignments/tree/main/web_scraping_assignments

**Comment by AlanK3** [comment]
https://github.com/AlanK3/is310-coding-assignments/tree/main/web-scraping-assignments
